### PR TITLE
Grabbing supplementary data from articles

### PR DESCRIPTION
**Pull request by willpearse**
...this is the pull request for issue #61.

Sorry for the delay; 'tis the job season to be jolly, etc...

I've refactored etc. to make everything look a bit more like the rest of the package. Let me know what you think; I won't in any way be offended if you don't want it in. There isn't perfect overlap between what you can get SI for and what you can get text for; I plan to get around to fixing that (...) but I reasoned you would rather have something than nothing!

I've made some changes to the .Rmd files as well, but for some reason I was having a lot of trouble building them (some sort of pandocs problem getting access to a server). I'm a Linux person, so I'm certain this reflects something weird that I've done.

### CODE REVIEW

**Comment by sckott** [comment]
Awesome!  Thanks. 

On vacation now.  Back next monday. 

One thing I see so far.  Can you use httr instead of rcurl? Don't want to introduce new  dependencies.

**Comment by sckott** [comment]
looking at this now

File: R/ft_get_si.R [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+#' (default) this will be a combination of the DOI and SI number
+#' @param dir directory to save file to. If \code{NULL} (default) this
+#' will be a temporary directory created for your files
+#' @param cache if \code{TRUE} (default), the file won't be downloaded
+#' again if it already exists (in a temporary directory creates, or
+#' your chosen \code{dir})
+#' @param vol Article volume (Proceedings journals only)
+#' @param issue Article issue (Proceedings journals only) 
+#' @author Will Pearse
+#' @examples
```
Comment (sckott) [inline R/ft_get_si.R]: please wrap examples in `\dontrun{}`

File: R/ft_get_si.R [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+#' #Put the function wherever you would put a file path
+#' crabs <- read.csv(ft_get_si("10.6084/m9.figshare.979288", 2))
+#'
+#' #ESA data papers and regular articles *must* be marked
+#' fungi <- read.csv(ft_get_si("E093-059", "myco_db.csv",
+#'                                         "esa_archives"))
+#' mammals <- read.delim(ft_get_si("E092-201", "MCDB_communities.csv",
```
Comment (sckott) [inline R/ft_get_si.R]: I think you want `read.delim(..., sep = ",")` here.

File: R/ft_get_si.R [hunk 3]
```diff
@@ -0,0 +1,240 @@
+#' Download supplementary materials from journals
+#'
+#' Download supplementary materials from papers. Put a call to this
+#' function where you would put a file-path - everything is cached by
+#' default, so you don't have to worry about multiple downloads in the
+#' same session.
+#' @param doi DOI of article. Note: if using ESA journal, this must be
```
Comment (sckott) [inline R/ft_get_si.R]: Please add the expected input to each parameter (e.g., `character`)

File: R/ft_get_si.R [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+#' @param save.name a name for the file to download. If \code{NULL}
+#' (default) this will be a combination of the DOI and SI number
+#' @param dir directory to save file to. If \code{NULL} (default) this
+#' will be a temporary directory created for your files
+#' @param cache if \code{TRUE} (default), the file won't be downloaded
+#' again if it already exists (in a temporary directory creates, or
+#' your chosen \code{dir})
+#' @param vol Article volume (Proceedings journals only)
+#' @param issue Article issue (Proceedings journals only) 
+#' @author Will Pearse
```
Comment (sckott) [inline R/ft_get_si.R]: give your email address if you like...

File: R/ft_get_si.R [hunk 5]
```diff
@@ -0,0 +1,240 @@
+#' Download supplementary materials from journals
```
Comment (sckott) [inline R/ft_get_si.R]: A `@details` section might be nice in this man file to explain this function a bit, and to state what publishers are available via this fxn, and what will come down the line

File: tests/testthat/test-ft_get_si.r [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  expect_true(file.exists(ft_get_si("E093-059", "myco_db.csv", "esa_archives")))
+  expect_true(file.exists(ft_get_si("E092-201", "MCDB_communities.csv", "esa_data_archives")))
+  expect_true(file.exists(ft_get_si("10.1126/science.1255768", "Appendix_BanksLeite_etal.txt")))
+  expect_true(file.exists(ft_get_si("10.1098/rspb.2015.0338", vol=282, issue=1811, 1)))
+})
+
+test_that("ft_get_si fails well", {
+  skip_on_cran()
+
+  expect_error(ft_get_si('nonsense', 1), "Error in get_si_pub(doi) : Cannot find publisher for DOI: nonsense")
```
Comment (sckott) [inline tests/testthat/test-ft_get_si.r]: in this `ft_get_si('nonsense', 1)` I get `Error in links[[1]] : subscript out of bounds`, not the expected output

File: R/ft_get_si.R [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+#' Download supplementary materials from papers. Put a call to this
+#' function where you would put a file-path - everything is cached by
+#' default, so you don't have to worry about multiple downloads in the
+#' same session.
+#' @param doi DOI of article. Note: if using ESA journal, this must be
+#' the ESA-specific article code (e.g., E092-201).
+#' @param si number of the supplment to be downloads (1, 2, 3, etc.),
+#' or (for ESA and Science journals) the name of the supplment (e.g.,
+#' "S1_data.csv").
+#' @param from Publisher of article. Optional, except for ESA journals
```
Comment (sckott) [inline R/ft_get_si.R]: it might be helpful to have more information here, e.g. a link for each publisher, could make a separate list in `@details`

**Comment by willpearse** [comment]
Hopefully this is everything you asked for!

**Comment by sckott** [comment]
thanks :+1:

**Comment by sckott** [comment]
@willpearse that one test is still failing, can you fix that

**Comment by willpearse** [comment]
The problem stems from my output from ````rcrossref::cr_works```` being different over the last few days. I can just refactor to fix this, but I'm concerned that I'm just confused about how this function works. Why is it that when I run the following:

````
cr_works(dois="not-a-DOI")
````

...I get something back? When I first wrote this I'm sure 'nonsense' DOIs returned with no data whatsoever, and the wrong result I get has changed in the last 24 hours.I wrote it, initially, to give two different error messages - one for when I couldn't find the right publisher, and the other for when I found the publisher but couldn't download from them.

Has something shifted on the crossref API, or am I just being stupid (I sense it's the latter!)

**Comment by sckott** [comment]
@willpearse this is a bug in the Crossref API, 

This is the API call from your e.g., http://api.crossref.org/works?query=not-a-DOI which gives that same result. 

Here's a related issue https://github.com/CrossRef/rest-api-doc/issues/37 in the repo for their API

Hopefully they'll get that fixed soon. 

So I guess a change in your test to account for that would be good.

**Comment by willpearse** [comment]
Done. Thanks for showing me the bug!

**Comment by willpearse** [comment]
:+1: thanks!

**Comment by sckott** [comment]
thank you for this!

**Pull request merged**